Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.cpp

```diff
@@ -114,4 +108,20 @@ Ref<StorageNamespace> WebStorageNamespaceProvider::createTransientLocalStorageNa
     return StorageNamespaceImpl::createSessionStorageNamespace(quota, sessionID);
 }
 
+RefPtr<StorageNamespace> WebStorageNamespaceProvider::sessionStorageNamespace(SecurityOrigin& topLevelOrigin, Page& page, ShouldCreateIfNotExists createIfNotExists)
+{
+    ASSERT(m_sessionStorageQuota != WebCore::StorageMap::noQuota);
+
+    auto& slot = m_sessionStorageNamespaces.add({ &page, topLevelOrigin.data() }, nullptr).iterator->value;
+    if (!slot && createIfNotExists == ShouldCreateIfNotExists::Yes)
+        slot = StorageNamespaceImpl::createSessionStorageNamespace(m_sessionStorageQuota, page.sessionID()).ptr();
+    return slot;
+}
+
+void WebStorageNamespaceProvider::setSessionStorageNamespace(WebCore::SecurityOrigin& topLevelOrigin, WebCore::Page& page, RefPtr<WebCore::StorageNamespace>&& newNamespace)
+{
+    auto& slot = m_sessionStorageNamespaces.add({ &page, topLevelOrigin.data() }, nullptr).iterator->value;
```

Comment:
What's the purpose of doing this in 2 instructions instead of one, do you want to make sure you overwrite the existing value if the key already existing in the map? If so, you can do so in a single line:
`m_sessionStorageNamespaces.set({ &page, topLevelOrigin.data() }, WTFMove(newNamespace))`

Or if you don't need to overwrite the existing value, just:
`m_sessionStorageNamespaces.add({ &page, topLevelOrigin.data() }, WTFMove(newNamespace))`

Reply:
Done. This was mostly because I copied a pattern I saw in other places of the codebase, but I agree directly using set() and add() seem better.